### DISCUSSION TITLE
Applications similar or related to AutoKey

### DISCUSSION THREAD

**Discussion by josephj11** [autokey/autokey#775] [General]
This discussion  is a place to keep track of other applications that may be of interest to AutoKey users and potential AutoKey users.

Any mentions here are not intended as recommendations or endorsements of other projects.

**Comment by josephj11** [comment]
[Gnome MacOS Remap](https://github.com/petrstepanov/gnome-macos-remap)

A package to remap hotkeys to make it easier for Mac users running Linux to work without relearning all new hotkeys.

Just discovered this one. It's actually directly uses AutoKey!

They have started work on a [version](https://github.com/petrstepanov/gnome-macos-remap-wayland) for Wayland that does not use AutoKey. It is in an early development stage.

**Comment by josephj11** [comment]
[Espanso](https://espanso.org)

This is a cross platform application (Windows/Linux/Mac) which appears to offer functionality similar to AutoKey phrases and scripts. It also supports working with forms.

Espanso does not support hotkeys/shortcuts, but intends to add them.

There is a version of Espanso that works on Wayland.

There is a [utility](https://ahk-to-espanso.as93.net) that converts AutoHotKey scripts into Espanso.

**Comment by josephj11** [comment]
[Keysharp](https://bitbucket.org/mfeemster/keysharp/src/master/)

This is a limited rewrite of AutoHotKey for Linux. It is currently in initial development and not for end users.

**Comment by josephj11** [comment]
AutoHotKey

This is a Windows-only application, but is known to work with very limited functionality under wine/proton.

**Comment by Elliria** [comment]
Mac has a built-in text-expander that can be accessed by going to:

System Preferences -> Security & Privacy -> Privacy -> Accessibility -> TextExpander

**Comment by docelic** [comment]
Another reimplementation of AutoHotKey for Linux: https://github.com/phil294/AHK_X11

**Comment by tapmoo** [comment]
I switched to [xdotool](https://manpages.ubuntu.com/manpages/trusty/man1/xdotool.1.html) when I couldn't get autokey working on Mint.

**Comment by josephj11** [comment]
@tapmoo We have a number of Mint users, so it does work. You do have to make sure you are using X11 and not Wayland for your display manager while running AutoKey. You should be able to select that on your user login screen.

xdtool is great. I use it for a number of things. It can even be invoked from within an AutoKey script.

The big difference is that, by itself, xdotool does not monitor keystrokes and then perform actions. You have to invoke it yourself somehow whenever you want it to do something. AutoKey's claim to fame is that it's already running and all you have to do is trigger it with a hotkey or a trigger abbreviation. That allows you to have many scripts/actions at your fingertips without running each one explicitly (possibly having to break your workflow to do so).

**Comment by private-lock** [comment]
KDE Plasma Team [said](https://blogs.kde.org/2025/11/26/going-all-in-on-a-wayland-future/#what-about-automation) on automation in Nov 2025:

> Many tools can be used for automation in the Wayland session; for example [wl-copy/wl-paste](https://manpages.ubuntu.com/manpages/jammy/man1/wl-copy.1.html), [ydotool](https://github.com/ReimuNotMoe/ydotool), [kdotool](https://github.com/jinliu/kdotool), [kscreen-doctor](https://tldr.dendron.so/notes/linux.kscreen-doctor.html), and the [plasma-apply-*](https://discuss.kde.org/t/cli-commands-to-switch-between-theming-settings/15544) tools. Generally Plasma is extensible enough that you can add what’s still missing yourself, for example through [KWin scripts](https://develop.kde.org/docs/plasma/kwin/) or plugins.

(links added by me)

And one more: [Hawck](https://github.com/snyball/Hawck)

**Reply by josephj11** [reply]
Thanks. Some of those tools are new to me.  I don't envision anyone in the AutoKey community digging into KWin, but you never know.

**Comment by josephj11** [comment]
Sikuli was around for a long time and morphed into Sikulix and sort of died.

It has apparently been resurrected in 2026 as [Oculix](https://github.com/oculix-org).

I looked into Sikuli a long time ago, but found it a bit too dense to penetrate. Hopefully, that has been improved.

It seemed to focus (no pun intended) on purely visual detection - like visgrep() on steroids. But that's as far as I got with it.

If anyone is more familiar with it, please add some details here.
